osu.Game/Overlays/BeatmapSet/LeaderboardModSelector.cs

```diff
@@ -0,0 +1,149 @@
+﻿// Copyright (c) ppy Pty Ltd <contact@ppy.sh>. Licensed under the MIT Licence.
+// See the LICENCE file in the repository root for full licence text.
+
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics;
+using osu.Game.Rulesets.Mods;
+using osu.Framework.Bindables;
+using osu.Game.Rulesets;
+using osuTK;
+using osu.Game.Rulesets.UI;
+using osu.Framework.Input.Events;
+using osu.Game.Graphics.UserInterface;
+using osuTK.Graphics;
+using System;
+using System.Linq;
+using osu.Framework.Extensions.IEnumerableExtensions;
+
+namespace osu.Game.Overlays.BeatmapSet
+{
+    public class LeaderboardModSelector : CompositeDrawable
+    {
+        public readonly BindableList<Mod> SelectedMods = new BindableList<Mod>();
+
+        private RulesetInfo ruleset;
+
+        public RulesetInfo Ruleset
+        {
+            get => ruleset;
+            set
+            {
+                if (ruleset?.Equals(value) ?? false)
+                {
+                    DeselectAll();
+                    return;
+                }
+
+                ruleset = value;
+
+                SelectedMods.Clear();
+                modsContainer.Clear();
+
+                if (ruleset == null)
+                    return;
+
+                modsContainer.Add(new ModButton(new ModNoMod()));
+                modsContainer.AddRange(ruleset.CreateInstance().GetAllMods().Where(m => m.Ranked).Select(m => new ModButton(m)));
+
+                modsContainer.ForEach(button => button.OnSelectionChanged = selectionChanged);
+            }
+        }
+
+        private readonly FillFlowContainer<ModButton> modsContainer;
+
+        public LeaderboardModSelector()
+        {
+            AutoSizeAxes = Axes.Both;
+            InternalChild = modsContainer = new FillFlowContainer<ModButton>
+            {
+                Anchor = Anchor.Centre,
+                Origin = Anchor.Centre,
+                AutoSizeAxes = Axes.Both,
+                Direction = FillDirection.Full,
+                Spacing = new Vector2(4),
+            };
+        }
+
+        private void selectionChanged(Mod mod, bool selected)
+        {
+            if (selected)
+                SelectedMods.Add(mod);
+            else
+                SelectedMods.Remove(mod);
+
+            if (!SelectedMods.Any() && !IsHovered)
```

Comment:
why not just make a single `updateHighlighted` method which handles everything, rather than having three different implementations in different places?